xarray/conventions.py

```diff
@@ -441,10 +440,12 @@ def stackable(dim: Hashable) -> bool:
         if decode_coords in [True, "coordinates", "all"]:
             var_attrs = new_vars[k].attrs
             if "coordinates" in var_attrs:
-                coord_str = var_attrs["coordinates"]
-                var_coord_names = coord_str.split()
-                if all(k in variables for k in var_coord_names):
-                    new_vars[k].encoding["coordinates"] = coord_str
+                var_coord_names = [
+                    c for c in var_attrs["coordinates"].split() if c in variables
+                ]
+                # only decode if there is at least one variable matching a coordinate
+                if var_coord_names:
+                    new_vars[k].encoding["coordinates"] = var_attrs["coordinates"]
                     del var_attrs["coordinates"]
```

Comment:
Actually i think I prefer this. We always move "coordinates" over if `decode_coords is not False`. THat way it is predictable.

```suggestion
                new_vars[k].encoding["coordinates"] = var_attrs["coordinates"]
                del var_attrs["coordinates"]
                if var_coord_names:
```

Reply:
That means: `coordinates` are propagated to `encoding` if given in the source file. Yes, that's even simpler.

Reply:
Oh, that one test need fixing.